API/src/main/java/de/brandwatch/minianalytics/api/service/MentionService.java

```diff
@@ -0,0 +1,74 @@
+package de.brandwatch.minianalytics.api.service;
+
+import com.google.gson.Gson;
+import de.brandwatch.minianalytics.api.solr.model.Mention;
+import de.brandwatch.minianalytics.api.solr.repository.MentionRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+
+@Component
+public class MentionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MentionService.class);
+
+    private final MentionRepository mentionRepository;
+
+    private static final Gson gson = new Gson();
+
+    @Autowired
+    public MentionService(MentionRepository mentionRepository) {
+        this.mentionRepository = mentionRepository;
+    }
+
+    public List<Mention> getMentionsFromQueryID(String queryID, String startDate, String endDate) {
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+
+        if(!endDate.equals("") && !startDate.equals("")){
+            if(!validateDates(convertStringDateToInstant(startDate), convertStringDateToInstant(endDate))){
+                throw new DateTimeException("Dates are not in the right order");
+            }
+        }
+
+        if (startDate.equals("")) {
+            sb.append("*");
+        } else {
+            sb.append(convertStringDateToInstant(startDate));
+        }
+
+        sb.append(" TO ");
+
+        if (endDate.equals("")) {
+            sb.append("*");
```

Comment:
I don't think this is a good idea. Consider a Solr database that has >10 years worth of data (a certain company called Brandwatch might have that ;) ). You surely don't want to request 10 years worth of data just because the client has not used the data param. Instead you want sensible defaults (such as start_date=now()-24hrs, end_date=now()), and possibly also caps (such as not more than a year range). The latter is not necessary for you though, but I would definitely implement the defaults. This will also save you some ifs and elses in the service, because you will just assume the date range has been validated by the controller.